webrender/src/device/gfx/device.rs

```diff
@@ -1403,42 +1449,49 @@ impl<B: hal::Backend> Device<B> {
         );
     }
 
-    pub fn bind_per_draw_textures(&mut self) {
+    pub fn bind_per_draw_textures(&mut self, can_skip_bind: bool) {
         debug_assert!(self.inside_frame);
         assert_ne!(self.bound_program, INVALID_PROGRAM_ID);
+        let descriptor_group = self.bound_program.1;
+
+        if !can_skip_bind {
+            let per_draw_bindings = PerDrawBindings(
+                [
+                    self.bound_textures[0],
+                    self.bound_textures[1],
+                    self.bound_textures[2],
+                ],
+                [
+                    self.bound_sampler[0],
```

Comment:
I *think* we can get away with binding *just* a single sampler here and using it for all 3 input textures.
There is just a few cases where we use multiple input textures. One of them is YUV, where clearly there don't need to be 3 different samplers.

Reply:
I've checked it and that's really the case, and we can use the first sampler.

Reply:
Made this change in https://github.com/szeged/webrender/pull/350/commits/b0be810ff981417d41a2600b6a4cd74ae5f9a2af#diff-cdf03c96bd34cb98c2d09b0be7658232R1475